src/convertToObject.js

```diff
@@ -3,16 +3,54 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
   // write your code here
+
+  let array = sourceString.split(';');
+
+  if (array.length === 0) {
+    return {};
+  }
+
+  // let's remove unnecesary signs here (clearing array)
+  array = array.map((key) => {
+    return key.replace(/\n/g, '').trim();
+  }).filter(Boolean);
+
+  // creating key:value pairs in array
+  array = array.map((key) => {
+    return key.split(':');
+  });
+
+  // cleaning inside, this time triming each element
+  array = array.map((key) => {
+    const fixedKey = key.map((el) => el.trim());
+
+    return fixedKey;
+  });
```

Comment:
E.g. here you could use two methods on key ( .split(...).map(...) ) while mapping the whole array only one time